frontend/src/metabase/parameters/utils/date-formatting.ts

```diff
@@ -111,52 +114,68 @@ function getFilterTitle(filter: any[]) {
   return prefix + desc;
 }
 
-export function formatAllOptionsWidget(urlEncoded: string) {
+export function formatAllOptionsWidget(urlEncoded: string): string | null {
   if (urlEncoded == null) {
     return null;
   }
   const filter = dateParameterValueToMBQL(urlEncoded, noopRef);
-
   return filter ? getFilterTitle(filter) : null;
 }
 
 function parseDateRangeValue(value: string) {
   const [start, end] = (value || "").split(RANGE_SEPARATOR);
-  return { start, end };
+  return { start: moment(start, true), end: moment(end, true) };
```

Comment:
Had to enable strict parsing (`true` argument) everywhere, otherwise `past1days` gets parsed as `yyyy-MM` (as year 1) etc